pkg/etcdcli/cached_members.go

```diff
@@ -0,0 +1,123 @@
+package etcdcli
+
+import (
+	"context"
+	"go.etcd.io/etcd/api/v3/etcdserverpb"
+	"sync"
+	"time"
+)
+
+const refreshPeriod = time.Minute
+
+type CachedMembers struct {
+	client  EtcdClient
+	lock    sync.RWMutex
+	members memberHealth
+
+	lastRefresh time.Time
+}
+
+func NewMemberCache(client EtcdClient) AllMemberLister {
+	return &CachedMembers{
+		client:      client,
+		lock:        sync.RWMutex{},
```

Comment:
I thought we wanted to move this to a plain `sync.Mutex` after some discussion here?

Reply:
+1